inc/functions_user.php

```diff
@@ -173,13 +172,107 @@ function update_password($uid, $password, $salt="")
  * @param string $password The md5()'ed password.
  * @param string $salt The salt.
  * @return string The password hash.
+ * @deprecated deprecated since version 1.8.9 Please use other alternatives.
  */
 function salt_password($password, $salt)
 {
 	return md5(md5($salt).$password);
 }
 
 /**
+ * Salts a password based on a supplied salt.
+ *
+ * @param string $password The input password.
+ * @param string $salt The salt used by the MyBB algorithm.
+ * @param string $user (Optional) An array containing password-related data.
+ * @return string The password hash.
+ */
+function create_password_hash($password, $salt, $user = false)
+{
+	global $plugins;
+
+	$parameters = compact('password', 'salt', 'user');
+
+	if(!defined('IN_INSTALL') && !defined('IN_UPGRADE'))
+	{
+		$plugins->run_hooks('create_password_hash', $parameters);
+	}
+
+	if(is_string($parameters))
+	{
+		return $parameters;
```

Comment:
The plugin class doesn't handle hook return values well: https://github.com/mybb/mybb/blob/mybb_1808/inc/class_plugins.php#L142-L145
If a custom function returns `false`, the hook outputs the original `$arguments` variable - this leads to messy implementations one way or another.